File: problem40/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        List<Integer> levelSums = new ArrayList<>();
        levelSums = levelSumList(root);
        int i = 0, maxLevel = 0;
        while (i < levelSums.size()) {
            if (levelSums.get(i) > levelSums.get(maxLevel))
                maxLevel = i;
            i++;
        }
        return maxLevel+1;
    }
    private List<Integer> levelSumList(TreeNode root) {
        if (root == null)
            return new ArrayList<>();

        List<Integer> left = new ArrayList<>();
        left = levelSumList(root.left);

        List<Integer> right = new ArrayList<>();
        right = levelSumList(root.right);

        List<Integer> level = new ArrayList<>();
        level.add(root.val);

        int len = Math.min(left.size(), right.size());
        for (int i = 0; i < len; i++)
            level.add(left.get(i) + right.get(i));

        if (left.size() < right.size())
            level.addAll(level.size(), right.subList(left.size(), right.size()));

        else if (left.size() > right.size())
            level.addAll(level.size(), left.subList(right.size(), left.size()));
        
        return level;
    }
}

File: problem40/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        max_sum, curr_level, max_level, queue = root.val, 0, 1, [root]

        while queue:
            curr_level += 1
            n, curr_sum = len(queue), 0

            # Remove nodes of this level & add nodes of the next level
            for i in range(n):
                front = queue.pop(0)
                curr_sum += front.val
                if front.left: queue.append(front.left)
                if front.right: queue.append(front.right)
            
            # Update max_level if sum of current level is greater
            if curr_sum > max_sum:
                max_sum = curr_sum
                max_level = curr_level
        
        return max_level
